docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+          check if there is a match.
+        - If there is a match, uses the lister for ManagedCluster and
+          find the matching clusters using the label selector expression
+          for clusters.
+        - If there are matching clusters, add the names of the cluster
+          to a hashmap setting the name of the cluster as a key. Cluster
+          groups from different placements are merged together.
+    -  If there are no matching clusters, the worker returns without
+        actions and is ready to process other events from the queue.
+    -  If there are matching clusters:
+        - Wraps the object into a ManifestWork
+        - Adds a label to the ManifestWork that is used to track the
+          placement that caused the object to be delivered to one or
+          more clusters. The label contains both the placement name
+          (note that placement is cluster-scoped) and the WDS name. A
+          label is attached for each placement whose selector matches
+          the object. This way, when a placement is deleted or
+          updated it is possible to locate the associated ManifestWorks
+          for deletion or label removal (if more than one label is
+          present, as other placements may “own” the object).
+        - For each cluster:
+          -  Sets the manifest namespace == name of the cluster
+          -  Uses the client connected to the ITS to do a server-side
+              apply of the manifest into the namespace.
+        - Worker returns and is ready to pick other keys from the queue.
+
+There are other event flows, based on the object GVK and type of event.
+The following sections broadly describe these flows:
+
+#### Object Deleted 
+
+When an object is deleted from the WDS, the handler’s *DeleteFunc* is
+invoked. Since most objects do not have a finalizer (which prevents
+deletion from the API server until the controller that sets the
+finalizer does some cleanup and then removes the object), a shallow copy
+of the object is added to a field in the key before pushing it to the
+work queue. Then:
+
+- Worker pulls the key from the work queue
+
+- Flow continues the same way as in the create/update scenario, however
+  the deletedObject field in the key indicates that the object has been
+  deleted and that it needs to be removed from all clusters.
+
+- Running the selectors matching logic as in the create/delete scenario
+  once again produces a list of clusters.
+
+- Worker iterates over the list of clusters and deletes the manifest
+  work for the object from each namespace associated with the cluster.
+
+- Worker returns and is ready to pick other keys from the queue.
+
+#### Placement Created or Updated 
+
+Worker pulls key from queue; if it is a placement and it has not been
+deleted (deletion timestamp not set) it follows the following flow:
+
+- Iterates all GVK-indexed listers
+- For each lister
+  - List objects
+    - For each object
+      - Checks object placement label
```

Comment:
You mean the generally plural set of labels on the corresponding wrapped object in the ITS(ITC)? If so then (a) being explicit about that would be helpful, and (b) explicitly mentioning the read (GET or cache fetch) that deliver access to that label set would be helpful.

Reply:
I realized my description here was not accurate and updated to be more clear and consistent with the code. Pls. check after I push the update to the PR.